### DISCUSSION TITLE
Feature request: tab or arrow through browser options

### DISCUSSION THREAD

**Discussion by bricin** [will-stone/browserosaurus#442] [Ideas]
While I can set a hotkey for each browser, my ideal usage would be to Tab (and alt+tab) through the list (or arrow keys) for the desired browser.

Love the app.

**Comment by will-stone** [comment]
Hi @bricin 👋 
Nice! I like this idea as then it wouldn't be that <kbd>enter</kbd> (<kbd>space</kbd>) opens the favourite app, it instead opens the _focused_ app, which by default just so happens to _be_  the first/favourite app 😄 

Some initial thoughts to self (although feel free to discuss if anything piques your interest):

- How best to display an app as focused?
- First app should be focused on show - Focus should reset to first app on show or hide (or both, just to be sure).
- I would suggest using <kbd>shift+tab</kbd> for going backwards as this is consistent with browsers.
- Do I capture the tab key and manually focus, or simply rely on in-built browser focus functionality? The latter is probably more accessible: why reinvent the wheel and all that.
- I'm not sure about arrow keys yet as this would be custom behaviour over what is default browser behaviour. Maybe this could be a separate feature for later, mainly because this involves having to know what's "up" and "down" whereas at the moment the tiles are essentially all in one line that overflows.

**Comment by will-stone** [comment]
Tabbing through apps is now a feature of the latest Browserosaurus (v16). Thanks for the suggestion 🙂
